setup.py

```diff
@@ -68,5 +68,7 @@
         'scikit-learn>=0.23,<2.0',
         'dill~=0.3',
         'xgboost~=1.3',
+        # dialogues
+        'dialogues @ git+https://github.com/Mehrad0711/dialogues',
```

Comment:
supporting e2e dialogues will be part of the core library. we can't import this lazily since it will be imported on top of many files.
Optional libraries are those which are purely for human consumption. 
I've kept this as a git dependency for now to track master cause the library is not stable yet and we keep pushing bug fixes etc.
It will become a regular dependency once I do a newer release after I test the whole thing.